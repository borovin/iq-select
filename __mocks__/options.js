export default {
    a: 'z',
    b: 'y',
    c: 'x'
}