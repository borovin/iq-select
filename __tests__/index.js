import IQSelect from '../index';
import React from 'react';
import options from '../__mocks__/options';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

jest.mock('../styles.styl', () => {});

it('Should render component', () => {
    const component = renderer.create(<IQSelect options={options} label='Test label'/>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('Should add focus mod', () => {
    const component = mount(<IQSelect options={options} label='Test label' />);

    component.find('.IQSelect__input').simulate('focus');
    expect(component.find('.IQSelect').hasClass('IQSelect_focus_true')).toBe(true);
});

it('Should remove focus mod', () => {
    const component = mount(<IQSelect options={options} label='Test label' />);

    component.find('.IQSelect__input').simulate('focus');
    component.find('.IQSelect__input').simulate('blur');
    expect(component.find('.IQSelect').hasClass('IQSelect_focus_false')).toBe(true);
});

it('Should add hasValue mod', () => {
    const component = mount(<IQSelect options={options} value="a" label='Test label' />);

    expect(component.find('.IQSelect').hasClass('IQSelect_hasValue_true')).toBe(true);
});

it('Should set value on option click', () => {
    const component = mount(<IQSelect name="testSelect" options={options} value="a" label='Test label' />);

    component.find('.IQSelect__input').simulate('focus');
    component.find('.IQSelect__option[data-value="b"]').simulate('mouseDown');
    expect(component.instance().value).toBe('b');
});