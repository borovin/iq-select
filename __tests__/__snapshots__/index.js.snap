exports[`test Should render component 1`] = `
<div
  className="IQSelect IQSelect_hasValue_false IQSelect_mobile_false IQSelect_focus_false IQSelect_openTo_bottom undefined">
  <div
    className="IQSelect__field">
    <label
      className="IQSelect__label">
      Test label
    </label>
    <div
      className="IQSelect__value" />
    <input
      className="IQSelect__input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      type="text" />
    <select
      className="IQSelect__nativeSelect"
      name="default name"
      onChange={[Function]}
      value="__default">
      <option
        disabled={true}
        hidden={true}
        value="__default">
        Test label
      </option>
      <option
        value="c">
        x
      </option>
      <option
        value="b">
        y
      </option>
      <option
        value="a">
        z
      </option>
    </select>
  </div>
  <div
    className="IQSelect__dropdown">
    <div
      className="IQSelect__option"
      data-value="c"
      onMouseDown={[Function]}>
      <span
        className="IQSelect__optionHighlight">
        
      </span>
      x
    </div>
    <div
      className="IQSelect__option"
      data-value="b"
      onMouseDown={[Function]}>
      <span
        className="IQSelect__optionHighlight">
        
      </span>
      y
    </div>
    <div
      className="IQSelect__option"
      data-value="a"
      onMouseDown={[Function]}>
      <span
        className="IQSelect__optionHighlight">
        
      </span>
      z
    </div>
  </div>
</div>
`;
